src/platform/qt/scripting/AutorunScriptModel.cpp

```diff
@@ -86,34 +127,47 @@ bool AutorunScriptModel::moveRows(const QModelIndex& sourceParent, int sourceRow
 		return false;
 	}
 
-	if (sourceRow < 0 || destinationChild < 0) {
+	if (sourceRow < 0 || destinationChild < 0 || count <= 0) {
 		return false;
 	}
 
-	if (sourceRow >= m_scripts.size() || destinationChild >= m_scripts.size()) {
+	int lastSource = sourceRow + count - 1;
+
+	if (lastSource >= m_scripts.size() || destinationChild > m_scripts.size()) {
 		return false;
 	}
 
-	if (count > 1) {
-		LOG(QT, WARN) << tr("Moving more than one row at once is not yet supported");
+	if (sourceRow == destinationChild - 1) {
 		return false;
 	}
 
-	auto item = m_scripts.takeAt(sourceRow);
-	m_scripts.insert(destinationChild, item);
-	save();
+	bool ok = beginMoveRows(QModelIndex(), sourceRow, lastSource, QModelIndex(), destinationChild);
+	if (!ok) {
+		qWarning() << tr("An error occurred while moving rows");
+		return false;
+	}
+	if (destinationChild < sourceRow) {
+		sourceRow = lastSource;
+	} else {
+		destinationChild -= 1;
+	}
+	for (int i = 0; i < count; i++) {
+		m_scripts.move(sourceRow, destinationChild);
+	}
+	endMoveRows();
+	emitScriptsChanged();
 	return true;
 }
 
 void AutorunScriptModel::addScript(const QString& filename) {
 	beginInsertRows({}, m_scripts.count(), m_scripts.count());
 	m_scripts.append(ScriptInfo { filename, true });
 	endInsertRows();
-	save();
+	emitScriptsChanged();
 }
 
-QList<QString> AutorunScriptModel::activeScripts() const {
```

Comment:
wow why the heck did I use `QList<QString>` here?

Reply:
I imagine because you were thinking of it parallel to `QList<ScriptInfo>`.